<a href="https://colab.research.google.com/github/ishneha1/AI_CourseWork/blob/main/WorksheetSignificance.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
from scipy import stats
from pyspark.sql import *
from pyspark.ml.feature import VectorAssembler
from pyspark.ml.regression import LinearRegression
from pyspark.sql.functions import col,desc
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
spark = SparkSession.builder.appName("WorksheetSignificance").getOrCreate()

In [ ]:
state_df= spark.read.csv('/content/state-0ee0ab38-0e53-40a9-82b2-eee1cb68eede.csv',header= True,inferSchema=True)
state_df.show(5)

+----------+------+----------+--------+------+-------+-----+------+
|Population|Income|Illiteracy|Life Exp|Murder|HS Grad|Frost|  Area|
+----------+------+----------+--------+------+-------+-----+------+
|      3615|  3624|       2.1|   69.05|  15.1|   41.3|   20| 50708|
|       365|  6315|       1.5|   69.31|  11.3|   66.7|  152|566432|
|      2212|  4530|       1.8|   70.55|   7.8|   58.1|   15|113417|
|      2110|  3378|       1.9|   70.66|  10.1|   39.9|   65| 51945|
|     21198|  5114|       1.1|   71.71|  10.3|   62.6|   20|156361|
+----------+------+----------+--------+------+-------+-----+------+
only showing top 5 rows


In [ ]:
state_pd=state_df.toPandas()
state_pd.head()

Population  Income  Illiteracy  Life Exp  Murder  HS Grad  Frost    Area
0        3615    3624         2.1     69.05    15.1     41.3     20   50708
1         365    6315         1.5     69.31    11.3     66.7    152  566432
2        2212    4530         1.8     70.55     7.8     58.1     15  113417
3        2110    3378         1.9     70.66    10.1     39.9     65   51945
4       21198    5114         1.1     71.71    10.3     62.6     20  156361

In [ ]:
state_pd.shape
#shows number of rows and columns

(50, 8)

** Hypothesis Testing

In [ ]:
#test whether the mean murder rate is 8.5
murder_data=state_pd['Murder'].values
murder_data

array([15.1, 11.3,  7.8, 10.1, 10.3,  6.8,  3.1,  6.2, 10.7, 13.9,  6.2,
        5.3, 10.3,  7.1,  2.3,  4.5, 10.6, 13.2,  2.7,  8.5,  3.3, 11.1,
        2.3, 12.5,  9.3,  5. ,  2.9, 11.5,  3.3,  5.2,  9.7, 10.9, 11.1,
        1.4,  7.4,  6.4,  4.2,  6.1,  2.4, 11.6,  1.7, 11. , 12.2,  4.5,
        5.5,  9.5,  4.3,  6.7,  3. ,  6.9])

In [ ]:
np.mean(murder_data)

np.float64(7.377999999999999)

In [ ]:
t_stat=(np.mean(murder_data)-8.5) / (np.std(murder_data,ddof=1) / np.sqrt(len(murder_data)))

In [ ]:
df=len(murder_data)

In [ ]:
p_value=2*stats.t.sf(np.abs(t_stat),df)

In [ ]:
ci=stats.t.interval(0.95,df,
                    loc=np.mean(murder_data),
                    scale=np.std(murder_data,ddof=1)/np.sqrt(len(murder_data))
)

In [ ]:
print("Mean:",np.mean(murder_data))
print("Sample Standard Deviation:",np.std(murder_data))

Mean: 7.377999999999999
Sample Standard Deviation: 3.6544378500666825


In [ ]:
print("Test Statistics:",t_stat)

Test Statistics: -2.1491677577323403


t_stat = 2.149 This means : The Sample mean -2.149 standard errors away from 8.5 Negative sign means the sample mean is Below 8.5 The larger the |t|, the more unusual our sample is

|t| = 0.5 ---> Very close to hypothesized mean (not unusual) |t| = 1.5 ---> Moderately far from hypothesized mean |t| = 2.1 ---> Far from hypothesized np.mean |t| = 5.0 ---> Very far from hypothesized mean


In [ ]:
print("Confidence Interval",ci)

Confidence Interval (np.float64(6.3294065080918065), np.float64(8.426593491908193))


In [ ]:
print("p-value",p_value)

p-value 0.03648613377210172


In [ ]:
print(f"we {'REJECT HO' if p_value <= 0.05 else 'FAIL TO REJECT HO'}")

we REJECT HO
